dev/com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/jaxrs/utils/FormUtils.java

```diff
@@ -0,0 +1,303 @@
+import org.apache.cxf.message.MessageUtils;
+import org.apache.cxf.phase.PhaseInterceptorChain;
+import org.apache.cxf.transport.http.AbstractHTTPDestination;
+
+public final class FormUtils {
+    public static final String FORM_PARAMS_FROM_HTTP_PARAMS = "set.form.parameters.from.http.parameters";
+    public static final String FORM_PARAM_MAP = "org.apache.cxf.form_data";
+    
+    private static final Logger LOG = LogUtils.getL7dLogger(FormUtils.class);
+    private static final String MULTIPART_FORM_DATA_TYPE = "form-data";  
+    private static final String MAX_FORM_PARAM_COUNT = "maxFormParameterCount";  
+    private static final String CONTENT_DISPOSITION_FILES_PARAM = "files";    
+    private FormUtils() {
+        
+    }
+    
+    public static String formToString(Form form) {
+        try (ByteArrayOutputStream bos = new ByteArrayOutputStream()) {
+            FormUtils.writeMapToOutputStream(form.asMap(), bos, StandardCharsets.UTF_8.name(), false);
+            return bos.toString(StandardCharsets.UTF_8.name());
+        } catch (Exception ex) {
+            // will not happen
+        }
+        return "";
+    }
+    
+    public static void restoreForm(FormEncodingProvider<Form> provider, 
+                                   Form form, 
+                                   Message message)
+        throws Exception {
+        CachedOutputStream os = new CachedOutputStream();
+        writeForm(provider, form, os);
+        message.setContent(InputStream.class, os.getInputStream());
+    }
+    
+    public static void writeForm(FormEncodingProvider<Form> provider, 
+                                 Form form, OutputStream os)
+        throws Exception {
+        provider.writeTo(form, Form.class, Form.class, new Annotation[]{}, 
+                         MediaType.APPLICATION_FORM_URLENCODED_TYPE, new MetadataMap<String, Object>(), os);
+    }
+    
+    public static Form readForm(FormEncodingProvider<Form> provider, Message message) 
+        throws Exception {
+        return provider.readFrom(Form.class, Form.class, 
+                              new Annotation[]{}, MediaType.APPLICATION_FORM_URLENCODED_TYPE, 
+                              new MetadataMap<String, String>(), 
+                              message.getContent(InputStream.class));
+    }
+    
+    public static void addPropertyToForm(MultivaluedMap<String, String> map, String name, Object value) {
+        if (!"".equals(name)) {
+            map.add(name, value.toString());
+        } else {
+            MultivaluedMap<String, Object> values = 
+                InjectionUtils.extractValuesFromBean(value, "");
+            for (Map.Entry<String, List<Object>> entry : values.entrySet()) {
+                for (Object v : entry.getValue()) {
+                    map.add(entry.getKey(), v.toString());
+                }
+            }
+        }
+    }
+    
+    public static String readBody(InputStream is, String encoding) {
```

Comment:
This will need a `@FFDCIgnore` annotation to avoid a new FFDC being printed here.